app/assets/javascripts/compute_resource.js

```diff
@@ -48,13 +49,16 @@ function testConnection(item) {
         $("#compute_resource_password").prop('disabled', false);
       }
       if (!/alert-danger/i.test(result)) {
-        notify("<p>" + __("Test connection was successful") + "</p>", 'success')
+        notify("<p>" + __("Test connection was successful") + "</p>", 'success');
+      } else {
+        $("#btn_datacenter").removeClass("btn-default").addClass("btn-danger");
```

Comment:
That's a refactoring, I'm not sure how that's related to the presence of the CSS class.  Even if it's rendering the whole form, the alert-danger CSS class could be present?